pkg/secrets-store/nodeserver.go

```diff
@@ -173,60 +174,72 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 			return nil, fmt.Errorf("Providers volume path not found. Set PROVIDERS_VOLUME_PATH")
 		}
 		if _, err := os.Stat(fmt.Sprintf("%s/%s/provider-%s", providerVolumePath, providerName, providerName)); err != nil {
-			glog.Errorf("failed to find provider %s, err: %v", providerName, err)
+			log.Errorf("failed to find provider %s, err: %v", providerName, err)
 			return nil, err
 		}
+
 		parametersStr, err := json.Marshal(parameters)
 		if err != nil {
-			glog.V(0).Infof("failed to marshal parameters, err: %v", err)
+			log.Errorf("failed to marshal parameters, err: %v", err)
 			return nil, err
 		}
 		secretStr, err := json.Marshal(secrets)
 		if err != nil {
-			glog.V(0).Infof("failed to marshal secrets, err: %v", err)
+			log.Errorf("failed to marshal secrets, err: %v", err)
 			return nil, err
 		}
 		permissionStr, err := json.Marshal(permission)
 		if err != nil {
-			glog.V(0).Infof("failed to marshal file permission, err: %v", err)
+			log.Errorf("failed to marshal file permission, err: %v", err)
 			return nil, err
 		}
 
-		glog.Infof("Calling provider: %s", providerName)
-		glog.Infof("provider command invoked: %s %s %s %s %s %s %s %s %s",
-			fmt.Sprintf("%s/%s/provider-%s", providerVolumePath, providerName, providerName),
-			"--attributes", "[REDACTED]",
-			"--secrets", "[REDACTED]",
-			"--targetPath", string(targetPath),
-			"--permission", string(permissionStr))
-
-		out, err := exec.Command(
-			fmt.Sprintf("%s/%s/provider-%s", providerVolumePath, providerName, providerName),
-			"--attributes", string(parametersStr),
+		log.Infof("Calling provider: %s", providerName)
+		providerBinary := fmt.Sprintf("%s/%s/provider-%s", providerVolumePath, providerName, providerName)
+		args := []string{"--attributes", string(parametersStr),
 			"--secrets", string(secretStr),
 			"--targetPath", string(targetPath),
-			"--permission", string(permissionStr),
-		).Output()
+			"--permission", string(permissionStr)}
+
+		// set log level for provider based on the value set for driver
+		if log.GetLevel() == log.DebugLevel {
+			args = append(args, "--debug", "true")
```

Comment:
This needs to be backward compatible with older provider binaries. `unknown flag: --debug`